microcosm_pubsub/tests/test_result.py

```diff
@@ -0,0 +1,139 @@
+        def func(message):
+            return True
+
+        result = MessageHandlingResult.invoke(
+            func=func,
+            message=self.message,
+        )
+
+        assert_that(
+            result,
+            has_properties(
+                media_type="application/vnd.microcosm.derived",
+                result=MessageHandlingResultType.SUCCEEDED,
+            ),
+        )
+        # ack
+        self.graph.sqs_consumer.sqs_client.delete_message.assert_called_with(
+            QueueUrl="queue",
+            ReceiptHandle=RECEIPT_HANDLE,
+        )
+
+    def test_skipped_falsey(self):
+        def func(message):
+            return False
+
+        result = MessageHandlingResult.invoke(
+            func=func,
+            message=self.message,
+        )
+
+        assert_that(
+            result,
+            has_properties(
+                media_type="application/vnd.microcosm.derived",
+                result=MessageHandlingResultType.SKIPPED,
+            ),
+        )
+        # ack
+        self.graph.sqs_consumer.sqs_client.delete_message.assert_called_with(
+            QueueUrl="queue",
+            ReceiptHandle=RECEIPT_HANDLE,
+        )
+
+    def test_skipped_exception(self):
+        def func(message):
+            raise SkipMessage("ignorance is bliss")
+
+        result = MessageHandlingResult.invoke(
+            func=func,
+            message=self.message,
+        )
+
+        assert_that(
+            result,
+            has_properties(
+                media_type="application/vnd.microcosm.derived",
+                result=MessageHandlingResultType.SKIPPED,
+            ),
+        )
+        # ack
+        self.graph.sqs_consumer.sqs_client.delete_message.assert_called_with(
+            QueueUrl="queue",
+            ReceiptHandle=RECEIPT_HANDLE,
+        )
+
+    def test_retried_nack(self):
+        def func(message):
+            raise Nack(3)
+
+        result = MessageHandlingResult.invoke(
+            func=func,
+            message=self.message,
+        )
+
+        assert_that(
+            result,
+            has_properties(
+                media_type="application/vnd.microcosm.derived",
+                result=MessageHandlingResultType.RETRIED,
+            ),
+        )
+        # nack
+        self.graph.sqs_consumer.sqs_client.change_message_visibility.assert_called_with(
+            QueueUrl="queue",
+            ReceiptHandle=RECEIPT_HANDLE,
+            VisibilityTimeout=3,
+        )
+
+    def test_failed_exception(self):
+        def func(message):
+            raise Exception("FAIL")
+
+        result = MessageHandlingResult.invoke(
+            func=func,
+            message=self.message,
+        )
+
+        assert_that(
+            result,
+            has_properties(
+                media_type="application/vnd.microcosm.derived",
+                result=MessageHandlingResultType.FAILED,
+            ),
+        )
+        # no nack (XXX)
```

Comment:
The default backoff policy doesn't return a default timeout so we don't nack by default on error. This is not great.